sqlglot/dialects/snowflake.py

```diff
@@ -41,9 +41,12 @@ def _build_datetime(
 ) -> t.Callable[[t.List], exp.Func]:
     def _builder(args: t.List) -> exp.Func:
         value = seq_get(args, 0)
+        scale_or_fmt = seq_get(args, 1)
+
         int_value = value is not None and is_int(value.name)
+        int_scale_or_fmt = scale_or_fmt is not None and is_int(scale_or_fmt.name)
 
-        if isinstance(value, exp.Literal):
+        if isinstance(value, exp.Literal) or (value and scale_or_fmt):
```

Comment:
Also a nit here, would it be a bit more readable to swap the OR rhs here with `or len(args) == 2)`? This hints that we'd be treating the 2-arg version more directly, instead of having to understand the conjuction

Reply:
I intentionally implemented the condition this way, because mypy yells at the following `value.is_string` otherwise.